### PR TITLE
Add conversion script.

### PR DESCRIPTION
**Pull request by sebpuetz**
Installing the package now installs `ffp-convert` which is portable for mac, windows and linux.

The most complicated part here is probably the testing setup.

### CODE REVIEW

File: src/finalfusion/scripts/convert.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    parser.add_argument(
+        "-t",
+        "--to",
+        type=str,
+        choices=formats,
+        default="finalfusion",
+        help=f"Valid choices: {formats} Default: 'finalfusion'",
+        metavar="OUTPUT_FORMAT")
+    args = parser.parse_args()
+    embeds = Format(getattr(args, 'from')).load(args.input)
```
Comment (sebpuetz) [inline src/finalfusion/scripts/convert.py:47]: need to use `getattr` here because `from` is a keyword...

File: tests/conversion_integration.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  "${tmp_dir}/fifu_buckets_to_text.dims.txt" textdims
+
+input="${TESTDIR}/data/fasttext.bin"
+verify_all_conversions  "${input}" fasttext
+
+function finish() {
+  rm -rf "$tmp_dir"
+}
+
+trap finish EXIT
```
Comment (danieldk) [inline tests/conversion_integration.sh]: Don't you want to set the trap immediately after creating the temporary directory?

**Review by danieldk — CHANGES_REQUESTED** [review]

File: tests/conversion_integration.sh (hunk 2 above)
Comment (sebpuetz) [reply tests/conversion_integration.sh]: I moved it up, but behaviour with `trap` at the end worked as expected on failure:

```
++ mktemp -d /tmp/run_conversion.XXXXXX
+ tmp_dir=/tmp/run_conversion.HIYH5d
+ trap finish EXIT
+++ dirname tests/conversion_integration.sh
++ cd tests
++ pwd -P
+ TESTDIR=/data/python_projects/ffp-finalfusion/tests
+ input=/data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt
+ verify_all_conversions /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt textdims
+ local input=/data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt
+ local in_format=textdims
+ local out_path_prefix=/tmp/run_conversion.HIYH5d/textdims
+ convert_and_verify /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt textdims /tmp/run_conversion.HIYH5d/textdims_to_fifu.fifu finalfusion
+ ffp-convert /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt -f textdims /tmp/run_conversion.HIYH5d/textdims_to_fifu.fifu -t finalfusion
tests/conversion_integration.sh: line 15: ffp-convert: command not found
+ finish
+ rm -rf /tmp/run_conversion.HIYH5d
```

File: tests/conversion_integration.sh [hunk 3]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+set -eux
```
Comment (danieldk) [inline tests/conversion_integration.sh]: Do you need `-x`? Produces a lot of output that is not really interesting when things are successful? Maybe echo what is being tested in `convert_and_verify`?

Comment (sebpuetz) [reply tests/conversion_integration.sh]: That's mostly a relict from debugging, removed it and replaced by an `echo` with the `ffp-convert` command

File: tests/conversion_integration.sh (hunk 2 above)
Comment (danieldk) [reply tests/conversion_integration.sh]: For me, it doesn't:

```
patching script interpreter paths in tests/conversion_integration.sh
tests/conversion_integration.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/x7fr0bvnwvqvr3zg60di9jxvfwimcw7m-bash-4.4-p23/bin/bash"
++ mktemp -d /tmp/run_conversion.XXXXXX
+ tmp_dir=/tmp/run_conversion.y1K1vI
+++ dirname tests/conversion_integration.sh
++ cd tests
++ pwd -P
+ TESTDIR=/build/source/tests
+ input=/build/source/tests/data/embeddings.dims.txt
+ verify_all_conversions /build/source/tests/data/embeddings.dims.txt textdims
+ local input=/build/source/tests/data/embeddings.dims.txt
+ local in_format=textdims
+ local out_path_prefix=/tmp/run_conversion.y1K1vI/textdims
+ convert_and_verify /build/source/tests/data/embeddings.dims.txt textdims /tmp/run_conversion.y1K1vI/textdims_to_fifu.fifu finalfusion
+ ffp-convert /build/source/tests/data/embeddings.dims.txt -f textdims /tmp/run_conversion.y1K1vI/textdims_to_fifu.fifu -t finalfusion
tests/conversion_integration.sh: line 8: ffp-convert: command not found
builder for '/nix/store/gxjciq20a4mr8nsgbysdd6lzikl543kf-python3.7-ffp-0.0.0.drv' failed with exit code 127
```

Comment (danieldk) [reply tests/conversion_integration.sh]: > I moved it up, but behaviour with `trap` at the end worked as expected on failure:
> 
> ```
> ++ mktemp -d /tmp/run_conversion.XXXXXX
> + tmp_dir=/tmp/run_conversion.HIYH5d
> + trap finish EXIT

Note that you are testing the version here with trap at the top ;).

Comment (sebpuetz) [reply tests/conversion_integration.sh]: Oops. Well then it should be working properly now!

Comment (sebpuetz) [reply tests/conversion_integration.sh]: So the `-x` was useful for something after all ;)

File: tests/conversion_integration.sh (hunk 3 above)
Comment (danieldk) [reply tests/conversion_integration.sh]: Thanks! Could you echo to stderr? E.g.

```
echo "hello world" >&2
```

Comment (sebpuetz) [reply tests/conversion_integration.sh]: done

**Review by danieldk — APPROVED** [review]

**Pull request merged**